File: LongestPalindromicString.java
    
		// Time Complexity : O(N)
	// Space Complexity : O(N)
	// Did this code successfully run on Leetcode : yes
	// Any problem you faced while coding this :
	 //NO

class Solution {
    public int longestPalindrome(String s) {
        
        char[] chArray = s.toCharArray();
        HashMap<Character,Integer> map  = new HashMap<>();
        boolean oddCharacters =false;
        int length=0;
        for(int i = 0; i < chArray.length ; i++){
            if(!map.containsKey(chArray[i]))
                map.put(chArray[i],1);
            else
                map.put(chArray[i],map.get(chArray[i])+1);
            
        }
         for (Map.Entry<Character,Integer> entry : map.entrySet()) {
             if(entry.getValue() % 2 == 0)
                 length += entry.getValue();
             else{
                 if(entry.getValue() >2 )
                     length += entry.getValue()-1;
                     oddCharacters = true;
             }
    }
        
        if(oddCharacters ){
            length +=1;
        }
        return length;
    }
}

File: longest_palindromic_string.py
# Time Complexity: O(n)
# Space Complexity: O(1) Maximum - lowercase charecters(unique) - 26
# Solution using Hashset
# Code ran successfully on Leetcode
class Solution:
    def longestPalindrome(self, s: str) -> int:
        # Hash set to check the occurence
        counter = set()
        count = 0
        for ch in s:
            # if a charecter is already present, add 2 as 
            if ch in counter:
                count += 2
                # remove the key from set
                counter.remove(ch)
            else:
                # if no occurence, add the key to set
                counter.add(ch)
        # if any element is present, means that odd length 
        if len(counter) != 0:
            count +=1
        return count
s = Solution().longestPalindrome("abccccdd")
print(s)